src/app/app.component.ts

```diff
@@ -86,7 +96,55 @@ export class AppComponent extends withUnsubscribe() implements OnInit {
 
   protected onRouterOutlet(componentInstance: PageComponent | unknown): void {
     if (componentInstance || componentInstance === null) {
-      this.sharedRoute.pageComponentInstance = componentInstance as PageComponent;
+      this.sharedRoute.pageComponentInstance =
+        componentInstance as PageComponent;
     }
   }
 }
+
+@Injectable()
+export class PageTitleStrategy extends TitleStrategy {
+  public constructor(
+    private title: Title,
+    private config: ConfigService,
+  ) {
+    super();
+  }
+
+  private routerState: RouterStateSnapshot;
+
+  /**
+   * Recursively builds the title from the page route and its parent routes
+   *
+   * @param subRoute A page route to construct the title for
+   * @returns A title of the route and its parents constructed in the format " | parent | subRoute"
+   */
+  private buildHierarchicalTitle(subRoute: MenuRoute): string {
+    // If the page route has an explicit way to construct the title, use the title callback
+    // if there is no `title` callback defined in the menuRoute, use the category label as a fallback
+    const componentTitle = " | " + (
+      subRoute?.title ?
+        subRoute.title(this.routerState) :
+        // since category labels are not title cased (first letter after space capitalized), we need to title case them
+        // since explicit route titles commonly include model names which are case sensitive, explicit titles should not change casing
+        // e.g. Project name "Tasmanian wetlands" != "Tasmanian Wetlands" as the user has explicitly not capitalized "Wetlands"
+        titleCase(subRoute.label)
```

Comment:
two points:

- this is a complex multiline conditional; it will read better if you just break it into a few steps with an if/else
- you should probably catch and handle exceptions (and log the error on the console) thrown by the title function

Reply:
I have made the requested code changes

![image](https://user-images.githubusercontent.com/33742269/227423026-69e12f8f-a8c6-4c94-8b88-c7859ad9b9d9.png)